### PR TITLE
SurfacePosition

### PR DESCRIPTION
**Pull request by Douglasdc3**
Started by @IDBeswick 

Initial attempt at implementing SurfacePosition. Algorithm is mostly shared with AirbornePosition, so the common elements are moved to the PositionUpdate class, which maintains a cache of even/odd frames, decides whether Global or Local calculation can be used, and generally contains the complexity of the CPR system. Some further sanity checks are required, and also a way to choose between multiple possible solutions for lat/lon for Surface data.

### CODE REVIEW

File: src/main/java/aero/t2s/modes/CprPosition.java [hunk 1]
```diff
@@ -40,6 +55,41 @@ public double getLon() {
...
+        this.surface = surface;
+    }
+
+    public boolean getSurface() {
+        return this.surface;
+    }
+
+    public void validateSurface() {
+        // For SurfacePositions we get 8 possible solutions: 2x latitude, 4x longitude zones at 90 degree increments
+        // TODO Could use receiver/reference lat/lon to chose which possibile solution is closest... or build-up an average over data received to-date
```
Comment (Douglasdc3) [inline src/main/java/aero/t2s/modes/CprPosition.java:68]: Closest to the reference point for the first decoding then for the second decoding the reference point should change to the previous position. This will be I think the most complete solution

**Comment by Douglasdc3** [comment]
Small changes were made mainly pre calculating the track and velocity information. Then uses the pre-calculated values to assign them to the track object when available

**Comment by IDBeswick** [comment]
SurfacePositionUpdates should now be working correctly, and AirbornePositionUpdates now correctly work for both Global and Local updates. Tests have been provided, but may not cover all edge-cases.

**Comment by Douglasdc3** [comment]
Looking good @IDBeswick thanks for the contribution!
